File: 0006-zigzag-conversion/0006-zigzag-conversion.cs
public class Solution {
    public string Convert(string s, int numRows) {
        if (numRows == 1)return s;
        string ans = "";
        for (int r = 0; r < numRows; r++) {
            int idx = 2 * (numRows - 1);
            for (int j = r; j < s.Length; j += idx) {
                ans += s[j];
                if (r != 0 && r != numRows - 1 && j + idx - 2 * r < s.Length)
                    ans += s[j + idx - 2 * r];
            }
        }
    return ans;
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution(object):
    def convert(self, s, numRows):
        if numRows == 1: return s
        ans = ""
        for r in range(numRows):
            idx = 2 * (numRows - 1)
            for j in range(r, len(s), idx):
                ans += s[j]
                if r != 0 and r != numRows - 1 and j + idx - 2 * r < len(s):
                    ans += s[j + idx - 2 * r]

        return ans
  